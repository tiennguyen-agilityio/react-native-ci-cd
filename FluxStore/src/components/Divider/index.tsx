import {memo, useMemo} from 'react';
import {ColorValue, DimensionValue, View, ViewStyle} from 'react-native';

// Stores
import {useThemeStore} from '@/stores';

export interface DividerProps {
  width?: DimensionValue;
  height?: DimensionValue;
  color?: ColorValue;
}

const Divider = ({width = '100%', height = 1, color}: DividerProps) => {
  const {
    theme: {background},
  } = useThemeStore();

  const dividerStyle: ViewStyle = useMemo(
    () => ({
      backgroundColor: color || background.quinary,
      height,
      width,
    }),
    [background.quinary, color, height, width],
  );

  return <View style={dividerStyle} />;
};

export default memo(Divider);
