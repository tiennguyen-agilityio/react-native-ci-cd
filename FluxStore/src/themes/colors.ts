export const colors = {
  transparent: 'transparent',
  white: {
    200: '#f3f3f6',
    300: '#f4f5f6',
    400: '#fafafa',
    500: '#fff',
    600: '#fcfcfd',
    700: '#f3f3f3',
    800: '#f7f7f8',
    900: '#E7E8E9',
    A900: '#f8f8fa',
  },
  black: {
    50: '#33302eb3',
    100: '#33302e',
    200: '#3a2c27',
    300: '#343434',
    400: '#00000080',
    500: '#000',
    600: '#1d1f22',
    800: '#23262f',
    700: '#332218',
    A50: '#33302e26',
    A100: '#141416',
  },
  gray: {
    50: '#d6d6d6',
    100: '#cbcdd8',
    200: '#b1b5c3',
    300: '#a3a5ad',
    400: '#8a8a8f',
    500: '#9b9b9b',
    600: '#9d9d9d',
    700: '#777e90',
    800: '#353945',
    900: '#464447',
    A50: '#e6e8ec',
    A100: '#e6e8fc',
    A200: '#e8e8ec',
    A300: '#737680',
    A400: '#707070',
    A500: '#bebfc4',
    A600: '#8e8e8e',
    A700: '#d8d8d8',
    A800: '#e2e2e2',
  },
  red: {
    200: '#f08c7d',
    400: '#ff6e6e',
    500: '#981b1f',
  },
  pink: {
    900: '#ef466f',
  },
  green: {
    200: '#508a7b',
    500: '#5ece7b',
  },
};

export const colorLinearGradients = {
  gray: {
    500: ['#353945', '#777e90'],
  },
};
