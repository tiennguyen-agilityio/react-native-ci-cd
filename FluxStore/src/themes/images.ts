export const Images = {
  welcome: require('@/assets/images/backgrounds/welcome.png'),
};

export const Carousels = {
  special: require('@/assets/images/carousels/special.png'),
  favorite: require('@/assets/images/carousels/favorite.png'),
  relax: require('@/assets/images/carousels/relax.png'),
  autumn: require('@/assets/images/carousels/autumn.png'),
};

export const Banners = {
  collection: require('@/assets/images/banners/collection.png'),
  beauty: require('@/assets/images/banners/beauty.png'),
  fabulous: require('@/assets/images/banners/fabulous.png'),
  shirts: require('@/assets/images/banners/shirts.png'),
  dresses: require('@/assets/images/banners/dresses.png'),
};
