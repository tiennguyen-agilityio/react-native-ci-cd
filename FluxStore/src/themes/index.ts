export * from './colors';
export * from './typography';
export * from './metrics';
export * from './images';
export * from './theme';
