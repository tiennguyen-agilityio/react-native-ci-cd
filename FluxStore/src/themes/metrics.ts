import {Dimensions} from 'react-native';

import {Metrics} from '@/interfaces';

const {width, height} = Dimensions.get('window');

export const borderRadius = {
  xs: 8,
  sm: 10,
  md: 15,
  lg: 20,
  xl: 30,
  circle: '50%',
};

export const metrics: Metrics = {
  screenWidth: width,
  screenHeight: height,
  dimensions: {
    xs: 8,
    sm: 10,
    md: 15,
    lg: 20,
    xl: 25,
    xxl: 32,
  },
};
