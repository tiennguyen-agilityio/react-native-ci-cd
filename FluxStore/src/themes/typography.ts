export const fontFamilies = {
  productSans: {
    regular: 'ProductSansRegular',
    medium: 'ProductSansMedium',
    bold: 'ProductSansBold',
  },
  roboto: {
    regular: 'RobotoRegular',
  },
  nunito: {
    semiBold: 'NunitoSemiBold',
  },
};

export const fontWeights = {
  regular: 400,
  medium: 500,
  semiBold: 600,
  bold: 700,
} as const;

export const fontSizes = {
  '3xl': 40,
  '2xl': 26,
  xl: 25,
  lg: 24,
  md: 20,
  base: 18,
  sm: 16,
  xs: 14,
  xxs: 13,
  tiny: 12,
  micro: 11,
  mini: 10,
};

export const lineHeights = {
  xl: 48,
  lg: 34,
  md: 24,
  base: 20,
  sm: 17,
  xs: 16,
  tiny: 12,
};

export const letterSpacings = {
  xl: 1.4,
  lg: 0.06,
  md: 0.02,
  base: 0,
  sm: -0.05,
  xs: -0.12,
};
