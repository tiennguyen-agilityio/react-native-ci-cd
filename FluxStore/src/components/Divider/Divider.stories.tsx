import {Meta, StoryObj} from '@storybook/react';

import {colors} from '@/themes';
import Divider from '.';

const meta: Meta<typeof Divider> = {
  title: 'Components/Divider',
  component: Divider,
};

export default meta;

type Story = StoryObj<typeof Divider>;

export const Default: Story = {
  args: {},
};

export const Height: Story = {
  args: {
    height: 5,
    color: colors.green[500],
  },
};
