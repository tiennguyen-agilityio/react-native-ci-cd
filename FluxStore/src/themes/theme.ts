import {Theme} from '@/interfaces';
import {colors} from './colors';
import {fontFamilies} from './typography';

export const lightThemes: Theme = {
  transparent: 'transparent',
  primary: colors.black[500],
  secondary: colors.white[700],
  tertiary: colors.black[300],
  error: colors.red[500],
  success: colors.green[500],
  link: colors.green[200],
  fonts: {
    default: {
      bold: fontFamilies.productSans.bold,
      medium: fontFamilies.productSans.medium,
      regular: fontFamilies.productSans.regular,
    },
    primary: {
      bold: fontFamilies.productSans.bold,
      medium: fontFamilies.productSans.medium,
      regular: fontFamilies.productSans.regular,
    },
    secondary: {
      medium: fontFamilies.nunito.semiBold,
    },
    tertiary: {
      regular: fontFamilies.roboto.regular,
    },
  },
  text: {
    default: colors.black[500],
    light: colors.white[500],
    primary: colors.black[600],
    secondary: colors.gray[800],
    tertiary: colors.gray[700],
    quaternary: colors.gray.A300,
    quinary: colors.black[50],
    senary: colors.gray[500],
    septenary: colors.gray.A500,
    link: colors.green[200],
    error: colors.red[500],
    success: colors.green[500],
  },
  background: {
    default: colors.white[500],
    primary: colors.black[300],
    secondary: colors.gray[600],
    tertiary: colors.white.A900,
    quaternary: colors.white[400],
    quinary: colors.white[200],
    senary: colors.white[700],
    septenary: colors.gray.A800,
    input: colors.white[800],
    radio: colors.green[200],
    checkbox: colors.green[500],
    icon: colors.white[500],
  },
  border: {
    default: colors.black.A50,
    primary: colors.gray[50],
    secondary: colors.gray.A50,
    tertiary: colors.white[500],
    quaternary: colors.gray.A500,
    quinary: colors.green[500],
    senary: colors.black[400],
    error: colors.red[200],
  },
  icon: {
    default: colors.black[400],
    checkbox: colors.gray.A600,
    primary: colors.black[100],
    secondary: colors.gray.A200,
    tertiary: colors.gray[600],
  },
};

export const darkThemes: Theme = {
  transparent: 'transparent',
  primary: colors.white[500],
  secondary: colors.black[800],
  tertiary: colors.white[600],
  error: colors.red[500],
  success: colors.green[500],
  link: colors.green[200],
  fonts: {
    default: {
      bold: fontFamilies.productSans.bold,
      medium: fontFamilies.productSans.medium,
      regular: fontFamilies.productSans.regular,
    },
    primary: {
      bold: fontFamilies.productSans.bold,
      medium: fontFamilies.productSans.medium,
      regular: fontFamilies.productSans.regular,
    },
    secondary: {
      medium: fontFamilies.nunito.semiBold,
    },
    tertiary: {
      regular: fontFamilies.roboto.regular,
    },
  },
  text: {
    default: colors.white[500],
    light: colors.black[500],
    primary: colors.white[600],
    secondary: colors.gray.A100,
    tertiary: colors.gray[200],
    quaternary: colors.gray[200],
    quinary: colors.gray.A200,
    senary: colors.gray.A200,
    septenary: colors.gray.A50,
    link: colors.green[200],
    error: colors.red[500],
    success: colors.green[500],
  },
  background: {
    default: colors.black.A100,
    primary: colors.white[600],
    secondary: colors.black[800],
    tertiary: colors.black[800],
    quaternary: colors.black[800],
    quinary: colors.black[800],
    senary: colors.white[700],
    septenary: colors.gray[700],
    input: colors.gray.A400,
    radio: colors.green[200],
    checkbox: colors.green[500],
    icon: colors.white[600],
  },
  border: {
    default: colors.white[600],
    primary: colors.gray[200],
    secondary: colors.gray[800],
    tertiary: colors.white[500],
    quaternary: colors.white[300],
    quinary: colors.green[500],
    senary: colors.gray.A50,
    error: colors.red[200],
  },
  icon: {
    default: colors.gray.A50,
    checkbox: colors.gray[200],
    primary: colors.white[600],
    secondary: colors.gray.A500,
    tertiary: colors.gray[200],
  },
};
