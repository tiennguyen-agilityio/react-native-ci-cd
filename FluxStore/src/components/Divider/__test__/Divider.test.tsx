import {render} from '@testing-library/react-native';

import {colors} from '@/themes';
import Divider from '..';

describe('Divider', () => {
  it('should render default props', () => {
    const {toJSON} = render(<Divider />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with customize props', () => {
    const props = {
      width: 300,
      height: 5,
      color: colors.green[500],
    };
    const {toJSON} = render(<Divider {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
