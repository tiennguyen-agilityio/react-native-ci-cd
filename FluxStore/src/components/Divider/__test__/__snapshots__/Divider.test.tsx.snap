// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render default props 1`] = `
<View
  style={
    {
      "backgroundColor": "#f3f3f6",
      "height": 1,
      "width": "100%",
    }
  }
/>
`;

exports[`Divider should render with customize props 1`] = `
<View
  style={
    {
      "backgroundColor": "#5ece7b",
      "height": 5,
      "width": 300,
    }
  }
/>
`;
